serialization/internal/default_serializer.go

```diff
@@ -413,3 +413,19 @@ func (*GobSerializer) Write(output serialization.DataOutput, i interface{}) erro
 	output.WriteData(&Data{network.Bytes()})
 	return nil
 }
+
+type HazelcastJSONSerializer struct{}
+
+func (*HazelcastJSONSerializer) ID() (id int32) {
+	return JSONSerializationType
+}
+
+func (*HazelcastJSONSerializer) Read(input serialization.DataInput) (object interface{}, err error) {
+	obj, err := input.ReadByteArray()
+	return core.HazelcastJSON{JSONString: obj}, err
+}
+
+func (*HazelcastJSONSerializer) Write(output serialization.DataOutput, object interface{}) (err error) {
+	output.WriteByteArray(object.(core.HazelcastJSON).JSONString)
```

Comment:
We are serializing this as UTF on the java. this needs to be consistent.

Reply:
@sancar  updated.